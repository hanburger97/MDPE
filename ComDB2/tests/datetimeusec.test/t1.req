set timezone UTC
select cast("2016-04-26T100000.8" as datetime)
select cast("2016-04-26T100000.000008" as datetime)
select cast("2016-04-26T100000.123456" as datetimeus)
select cast(1461607247 as datetime)
select cast(1461607247.123 as datetime)
select cast("1461607247.123" as datetime)
select cast(1461607247.12345 as datetime)
select cast("1461607247.12345" as datetime)
select cast("2001-01-02" as datetime) + cast("10" as year)
select cast("2001-01-02" as datetime) + cast("10" as month)
select cast("2001-01-02" as datetime) + cast("10" as day)
select cast("2001-01-02" as datetime) + cast("10" as hour)
select cast("2001-01-02" as datetime) + cast("10" as minute)
select cast("2001-01-02" as datetime) + cast("10" as second)
select days(cast("2014-08-22" as datetimeus), "w")
select days(cast("2014-08-22" as datetimeus), "m")
select days(cast("2014-08-22" as datetimeus), "y")
select days(cast("2015-02-23" as datetimeus), "w")
select days(cast("2015-02-23" as datetimeus), "m")
select days(cast("2015-02-23" as datetimeus), "y")
select cast(0.001 as datetime)
select cast(0.000001 as datetime)
select cast(0.999 as second)
select cast(0.999999 as second)
select cast(1.000246 as second)
select cast(1.246246 as second)
select cast(1.246 as second)
select cast("1.000246" as second)
select cast("1.246246" as second)
select cast("1.246" as second)
select cast("1 02:04:06.000246" as second)
select cast("1 02:04:06.246" as second)
select cast("2016-04-26T100000.123" as datetime) - cast("2016-04-25T100000.123" as datetime)
select cast("2016-04-26T100000.123" as datetime) - cast("2016-04-25T100000.000123" as datetimeus)
select cast("2016-04-26T100000.123456" as datetimeus) - cast("2016-04-25T100000.123" as datetime)
select cast("2016-04-26T100000.123456" as datetimeus) - cast("2016-04-25T100000.123455" as datetimeus)
select cast("2016-04-26T100000.123456" as datetimeus) - cast(0.000010 as second)
select cast("2016-04-26T100000.123456" as datetime)   - cast(0.000010 as second)
select cast("2016-04-26T100000.123456" as datetime)   - cast(0.123 as second)
select cast("2016-04-26T100000.123456" as datetimeus) - cast(0.123 as second)
select cast("2016-04-26T100000.123456" as datetimeus) - cast("0.000010" as second)
select cast("2016-04-26T100000.123456" as datetimeus) - cast("0.123455" as second)
select cast("2016-04-26T100000.123456" as datetime)   - cast("0.000010" as second)
select cast("2016-04-26T100000.123456" as datetime)   - cast("0.123" as second)
select cast("2016-04-26T100000.123456" as datetimeus) - cast("0.123" as second)
select cast(0.000010 as second)    +   cast("2016-04-26T100000.123456" as datetimeus)
select cast(0.000010 as second)    +   cast("2016-04-26T100000.123456" as datetime)  
select cast(0.123 as second)       +   cast("2016-04-26T100000.123456" as datetime)  
select cast(0.123 as second)       +   cast("2016-04-26T100000.123456" as datetimeus)
select cast("0.000010" as second)  +   cast("2016-04-26T100000.123456" as datetimeus)
select cast("0.123455" as second)  +   cast("2016-04-26T100000.123456" as datetimeus)
select cast("0.000010" as second)  +   cast("2016-04-26T100000.123456" as datetime)  
select cast("0.123" as second)     +   cast("2016-04-26T100000.123456" as datetime)  
select cast("0.123" as second)     +   cast("2016-04-26T100000.123456" as datetimeus)
select cast(1 as second) - cast(0.005 as second)
select cast(1 as second) - cast(0.0005 as second)
select cast(1 as second) - cast(0.00005 as second)
select cast(1 as second) - cast(0.000005 as second)
select cast(0.999 as second) + cast(0.001 as second)
select cast(0.999 as second) + cast(0.0001 as second)
select cast(0.999 as second) + cast(0.00001 as second)
select cast(0.999 as second) + cast(0.000001 as second)
select cast(1.555555 as second) - cast(0.005 as second)
select cast(1.555555 as second) + cast(0.005 as second)
select cast(0.999999 as second) + cast(0.001 as second)
select cast(0.999999 as second) + cast(0.0001 as second)
select cast(0.999999 as second) + cast(0.00001 as second)
select cast(0.999999 as second) + cast(0.000001 as second)
select cast(0.246246 as second) + cast(0.001 as second)
select cast(0.246246 as second) + cast(0.0001 as second)
select cast(0.246246 as second) + cast(0.00001 as second)
select cast(0.246246 as second) + cast(0.000001 as second)
