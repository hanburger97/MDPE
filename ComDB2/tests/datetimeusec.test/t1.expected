(cast("2016-04-26T100000.8" as datetime)="2016-04-26T100000.008 UTC")
(cast("2016-04-26T100000.000008" as datetime)="2016-04-26T100000.000008 UTC")
(cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.123456 UTC")
(cast(1461607247 as datetime)="2016-04-25T180047.000 UTC")
(cast(1461607247.123 as datetime)="2016-04-25T180047.123 UTC")
(cast("1461607247.123" as datetime)="2016-04-25T180047.123 UTC")
(cast(1461607247.12345 as datetime)="2016-04-25T180047.123450 UTC")
(cast("1461607247.12345" as datetime)="2016-04-25T180047.123450 UTC")
(cast("2001-01-02" as datetime) + cast("10" as year)="2011-01-02T000000.000 UTC")
(cast("2001-01-02" as datetime) + cast("10" as month)="2001-11-02T000000.000 UTC")
(cast("2001-01-02" as datetime) + cast("10" as day)="2001-01-12T000000.000 UTC")
(cast("2001-01-02" as datetime) + cast("10" as hour)="2001-01-02T100000.000 UTC")
(cast("2001-01-02" as datetime) + cast("10" as minute)="2001-01-02T001000.000 UTC")
(cast("2001-01-02" as datetime) + cast("10" as second)="2001-01-02T000010.000 UTC")
(days(cast("2014-08-22" as datetimeus), "w")=5)
(days(cast("2014-08-22" as datetimeus), "m")=21)
(days(cast("2014-08-22" as datetimeus), "y")=233)
(days(cast("2015-02-23" as datetimeus), "w")=1)
(days(cast("2015-02-23" as datetimeus), "m")=22)
(days(cast("2015-02-23" as datetimeus), "y")=53)
(cast(0.001 as datetime)="1970-01-01T000000.001 UTC")
(cast(0.000001 as datetime)="1970-01-01T000000.000001 UTC")
(cast(0.999 as second)="0 00:00:00.999")
(cast(0.999999 as second)="0 00:00:00.999999")
(cast(1.000246 as second)="0 00:00:01.000246")
(cast(1.246246 as second)="0 00:00:01.246246")
(cast(1.246 as second)="0 00:00:01.246")
(cast("1.000246" as second)="0 00:00:01.000246")
(cast("1.246246" as second)="0 00:00:01.246246")
(cast("1.246" as second)="0 00:00:01.246")
(cast("1 02:04:06.000246" as second)="1 02:04:06.000246")
(cast("1 02:04:06.246" as second)="1 02:04:06.246")
(cast("2016-04-26T100000.123" as datetime) - cast("2016-04-25T100000.123" as datetime)="1 00:00:00.000")
(cast("2016-04-26T100000.123" as datetime) - cast("2016-04-25T100000.000123" as datetimeus)="1 00:00:00.122877")
(cast("2016-04-26T100000.123456" as datetimeus) - cast("2016-04-25T100000.123" as datetime)="1 00:00:00.000456")
(cast("2016-04-26T100000.123456" as datetimeus) - cast("2016-04-25T100000.123455" as datetimeus)="1 00:00:00.000001")
(cast("2016-04-26T100000.123456" as datetimeus) - cast(0.000010 as second)="2016-04-26T100000.123446 UTC")
(cast("2016-04-26T100000.123456" as datetime)   - cast(0.000010 as second)="2016-04-26T100000.123446 UTC")
(cast("2016-04-26T100000.123456" as datetime)   - cast(0.123 as second)="2016-04-26T100000.000456 UTC")
(cast("2016-04-26T100000.123456" as datetimeus) - cast(0.123 as second)="2016-04-26T100000.000456 UTC")
(cast("2016-04-26T100000.123456" as datetimeus) - cast("0.000010" as second)="2016-04-26T100000.123446 UTC")
(cast("2016-04-26T100000.123456" as datetimeus) - cast("0.123455" as second)="2016-04-26T100000.000001 UTC")
(cast("2016-04-26T100000.123456" as datetime)   - cast("0.000010" as second)="2016-04-26T100000.123446 UTC")
(cast("2016-04-26T100000.123456" as datetime)   - cast("0.123" as second)="2016-04-26T100000.000456 UTC")
(cast("2016-04-26T100000.123456" as datetimeus) - cast("0.123" as second)="2016-04-26T100000.000456 UTC")
(cast(0.000010 as second)    +   cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.123466 UTC")
(cast(0.000010 as second)    +   cast("2016-04-26T100000.123456" as datetime)="2016-04-26T100000.123466 UTC")
(cast(0.123 as second)       +   cast("2016-04-26T100000.123456" as datetime)="2016-04-26T100000.246456 UTC")
(cast(0.123 as second)       +   cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.246456 UTC")
(cast("0.000010" as second)  +   cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.123466 UTC")
(cast("0.123455" as second)  +   cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.246911 UTC")
(cast("0.000010" as second)  +   cast("2016-04-26T100000.123456" as datetime)="2016-04-26T100000.123466 UTC")
(cast("0.123" as second)     +   cast("2016-04-26T100000.123456" as datetime)="2016-04-26T100000.246456 UTC")
(cast("0.123" as second)     +   cast("2016-04-26T100000.123456" as datetimeus)="2016-04-26T100000.246456 UTC")
(cast(1 as second) - cast(0.005 as second)="0 00:00:00.995")
(cast(1 as second) - cast(0.0005 as second)="0 00:00:00.999500")
(cast(1 as second) - cast(0.00005 as second)="0 00:00:00.999950")
(cast(1 as second) - cast(0.000005 as second)="0 00:00:00.999995")
(cast(0.999 as second) + cast(0.001 as second)="0 00:00:01.000")
(cast(0.999 as second) + cast(0.0001 as second)="0 00:00:00.999100")
(cast(0.999 as second) + cast(0.00001 as second)="0 00:00:00.999010")
(cast(0.999 as second) + cast(0.000001 as second)="0 00:00:00.999001")
(cast(1.555555 as second) - cast(0.005 as second)="0 00:00:01.550555")
(cast(1.555555 as second) + cast(0.005 as second)="0 00:00:01.560555")
(cast(0.999999 as second) + cast(0.001 as second)="0 00:00:01.000999")
(cast(0.999999 as second) + cast(0.0001 as second)="0 00:00:01.000099")
(cast(0.999999 as second) + cast(0.00001 as second)="0 00:00:01.000009")
(cast(0.999999 as second) + cast(0.000001 as second)="0 00:00:01.000000")
(cast(0.246246 as second) + cast(0.001 as second)="0 00:00:00.247246")
(cast(0.246246 as second) + cast(0.0001 as second)="0 00:00:00.246346")
(cast(0.246246 as second) + cast(0.00001 as second)="0 00:00:00.246256")
(cast(0.246246 as second) + cast(0.000001 as second)="0 00:00:00.246247")
